File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
// 풀이1) 배열
// 풀이2) Map
function solution(s) {
    let lastSeen = new Map();
    let answer = [];
    
    for (let i = 0; i < s.length; i++) {
        if(!lastSeen.has(s[i])) {
            answer.push(-1);
        } else {
            answer.push(i - lastSeen.get(s[i]));
        }
        lastSeen.set(s[i], i);
    }
    
    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    last_seen = {}
    answer = []
    
    for i in range(len(s)):
        if s[i] not in last_seen:
            answer.append(-1)
        else:
            answer.append(i - last_seen[s[i]])
        last_seen[s[i]] = i
        
    return answer